### ISSUE TITLE
ERROR Malformed floating point number

### ISSUE THREAD

**Issue by rcgz** [OPM/opm-common#3042]
I have an error that does not allow the simulation to continue, the message is:
```
Error: Problem with keyword TIGHTEN
Internal error: Malformed floating point number 'RPTRST'
Error: Unrecoverable errors were encountered while loading input
```
Can you help me by telling what went wrong please

Thanks

**Comment by OPMUSER** [comment]
Please post a section of deck where the error is.

**Comment by rcgz** [comment]
`Error: Problem with keyword TIGHTEN

In /home/ubuntu/Escritorio/OPM/OPM Data/EN_2019.sim/HM_FP_POES_2020-09-02/HM_FP_POES_2020-09-02_SCH.INC line 16.

Internal error: Malformed floating point number 'SKIPREST'

Error: Unrecoverable errors were encountered while loading input`

the error is generated when reading data from the different .include files

**Comment by OPMUSER** [comment]
I need to see the actual section in the input file may be 10 lines above and below in order to help.

**Comment by rcgz** [comment]
sorry, from the .include file?

**Comment by OPMUSER** [comment]
Yes the input were the error occurs in the include file +/- 10 lines.

**Comment by bska** [comment]
>> ```
>> Error: Problem with keyword TIGHTEN
>> .../HM_FP_POES_2020-09-02_SCH.INC line 16.
>> Internal error: Malformed floating point number 'SKIPREST'
>> ```
>
> I need to see the actual section in the input file may be 10 lines above and below in order to help.

As @OPMUSER says, it's generally not feasible to diagnose problems like these without having access to at least a small section of the intput&mdash;in this case the `HM_FP_POES_2020-09-02_SCH.INC` include file&mdash;near where error occurs.  That said, and because I've seen similar symptoms ("malformed floating point number...") before, I'm going to guess that you've not properly terminated the `TIGHTEN` keyword.

This keyword needs to be entered as
```
TIGHTEN
  1.23 /
```
with a terminating `/` character following the single floating-point value, either on the same line or on the subsequent line.

**Comment by rcgz** [comment]
`RPTONLY

TUNING
1 0.5 1 0.15 3 0.3 0.4 1.5 1E20 /
0.1 0.001 5E-7 0.005 10 5 5E-6 0.05 0.001 0.025/
12 1 99 1 15 8 1E6 1E6 1E6 1E6/

TIGHTEN
1  /

SKIPREST                               -- Generated : Petrel

RPTSCHED                               -- Generated : Petrel
  FIP=2 WELLS /

RPTRST                                 -- Generated : Petrel
  BASIC=3 FLOWS FREQ /

SKIP                                   -- Generated : Petrel
--Hint: Select wells on the input tree, drop in with the blue arrow, then add rules with the rule pop-up

ENDSKIP                                -- Generated : Petrel

GRUPTREE                               -- Generated : Petrel
  'GROUP 1' FIELD /
  /

DATES                                  -- Generated : Petrel
  29 JUL 2003 /
  /
`
thanks for the help, this is the input file.

**Comment by rcgz** [comment]
`RPTONLY

TUNING
1 0.5 1 0.15 3 0.3 0.4 1.5 1E20 /
0.1 0.001 5E-7 0.005 10 5 5E-6 0.05 0.001 0.025/
12 1 99 1 15 8 1E6 1E6 1E6 1E6/

TIGHTEN
1 /

SKIPREST 

RPTSCHED 
FIP=2 WELLS /

RPTRST 
BASIC=3 FLOWS FREQ /

SKIP 

ENDSKIP 

GRUPTREE 
'GROUP 1' FIELD /
/

DATES 
29 JUL 2003 /
/
`
thanks for the help, this is the input file.

**Comment by OPMUSER** [comment]
This should work, hopefully:
```
RPTONLY

TUNING
1    0.5   1     0.15   3   0.3   0.4   1.5   1E20         /
                                                           / NOT SUPPORTED
                                                           / NOT SUPPORTED
--
-- NOT SUPPORTED
--
-- TIGHTEN
-- 1 /

SKIPREST

RPTSCHED
FIP=2 WELLS=2 /

RPTRST
BASIC=3  FREQ=1 /
--
-- THIS HAS NO EFFECT BECAUSE THERE IS NOTHING IN BETWEEN SO COMMENT OUT
-- SKIP

-- ENDSKIP

GRUPTREE
'GROUP 1' FIELD /
/

DATES
29 JUL 2003 /
/
```

A few things to note:

- Only the first line of the `TUNING` keyword is used if, and only if, you set `enable-tuning=true` in the parameter file; so I commented out the other two lines to avoid confusion.
- The `TIGHTEN` keyword is not supported, so I commented it out to avoid confusion.
- For  `RPTSCHED`, I suspect that the WELLS sub-keyword is the issue, as it should be of the same form as the FIP sub-keyword, that is` WELLS=2`.
-  Similarly for `RPTRST` and the `FREQ` sub-keyword and FLOWS sub-keyword is not supported according to the manual.

Try and be more neat in your input decks as it helps in catching errors and makes the deck more readable. If you download OPMRUN (https://github.com/OPM/opm-utilities/tree/master/opmrun) then there is a keyword editor that will generate the keywords for you in a similar format as the examples in the manual.

**Comment by rcgz** [comment]
ok, thanks a lot

**Comment by bska** [comment]
@rcgz: Are you satisfied that the problem has been resolved?  If so, please let us or or close the Issue yourself.  If not, please provide additional details of what problems you see?

**Comment by rcgz** [comment]
that problem is no longer there, but now when starting the report step it only reaches the middle and then generates an error:
<<Error: Unrecoverable errors were encountered while loading input>>

**Comment by bska** [comment]
> when starting the report step it only reaches the middle and then generates an error:
> ```
> Error: Unrecoverable errors were encountered while loading input
> ```

I see.  Thanks for the update.  In this case, I believe we'll need to see at least the full `.DBG` file from your failed simulation run, but it's likely that we'll need to see the actual input files too.  The above diagnostic is emitted when the input reader encounters something that it cannot make sense of.  It might be an unterminated keyword or record or it might be an unexpected number or type of items in an individual keyword.

I recently amended the diagnostic message (PR OPM/opm-simulators#3912) to include additional information that might aid the user in identifying the problem, but unless you'd like to build the simulator from source code yourself that facility won't be available to you until the 2022.10 release later this year.
